csrc/host_ir/jit.cpp

```diff
@@ -166,29 +178,388 @@ void printLlvmIr(llvm::Function* func, std::string_view msg) {
+    return builder.CreateUDiv(numerator, rhs_value);
+  } else {
+    NVF_THROW("LLVM Lowering Error: Unsupported binary operation type in extent calculation: ", binary_op->getBinaryOpType());
+  }
+  return nullptr;
+}
+
+llvm::Value* createValueForUnaryOp(UnaryOp* unary_op, std::unordered_map<Val*, llvm::Value*>& val_to_value, llvm::IRBuilder<>& builder) {
+  auto* in = unary_op->in()->as<Val>();
+  llvm::Value* in_value = getOrCreateValueForExtent(in, val_to_value, builder);
+  if(unary_op->getUnaryOpType() == UnaryOpType::Cast) {
+    return in_value;
+  } else if(unary_op->getUnaryOpType() == UnaryOpType::Abs) {
+    llvm::Value* is_negative = builder.CreateICmpSLT(in_value, builder.getInt64(0));
+    llvm::Value* negated = builder.CreateNeg(in_value);
+    return builder.CreateSelect(is_negative, negated, in_value);
+  } else if(unary_op->getUnaryOpType() == UnaryOpType::Neg) {
+    return builder.CreateNeg(in_value);
+  } else {
+    NVF_THROW("LLVM Lowering Error: Unsupported unary operation type in extent calculation: ", unary_op->getUnaryOpType());
+  }
+  return nullptr;
+}
+
+llvm::Value* createValueForExtent(
+    Val* val,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value,
+    llvm::IRBuilder<>& builder) {
+  if (val->isA<IterDomain>()) {
+    if(val->as<IterDomain>()->isBroadcast()) {
+      if (val->as<IterDomain>()->hasExpandedExtent()) {
+        return getOrCreateValueForExtent(val->as<IterDomain>()->expandedExtent(), val_to_value, builder);
+      }
+      return builder.getInt64(1);
+    } else {
+      return getOrCreateValueForExtent(val->as<IterDomain>()->extent(), val_to_value, builder);
+    }
+  } else if (val->isConst()) {
+    return builder.getInt64(val->value().as<int64_t>());
+  } else if (Expr* def = val->definition()) {
+    if (auto* binary_op = def->as<BinaryOp>()) {
+      return createValueForBinaryOp(binary_op, val_to_value, builder);
+    } else if (auto* unary_op = def->as<UnaryOp>()) {
+      return createValueForUnaryOp(unary_op, val_to_value, builder);
+    } else {
+      NVF_THROW(
+        "LLVM Lowering Error: createValueForExtent called with unsupported "
+        "expression type: ",
+        def->getOpString());
+    }
+  } else {
+    NVF_THROW(
+        "LLVM Lowering Error: createValueForExtent called with unfounded "
+        "val: ",
+        val->toString());
+  }
+  return nullptr;
+}
+
+llvm::Value* getOrCreateValueForExtent(Val* extent, std::unordered_map<Val*, llvm::Value*>& val_to_value, llvm::IRBuilder<>& builder) {
+  auto it = val_to_value.find(extent);
+  if (it != val_to_value.end()) {
+    return it->second;
+  }
+  llvm::Value* value = createValueForExtent(extent, val_to_value, builder);
+  // after recursive call, the original iterator may no longer be valid
+  val_to_value[extent] = value;
+  return value;
+}
+
+// Helper function to map current domain to input domain, 
+// boundary_vals tracks which high level domain has been visited
+Val* mapToInputDomain(
```

Comment:
nitpick, the return value doesn't seem to be used at all. maybe return a boolean instead.